runtime/src/iree/hal/drivers/hip/per_device_information.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_HAL_DRIVERS_HIP_PER_DEVICE_INFORMATION_H__
+#define IREE_HAL_DRIVERS_HIP_PER_DEVICE_INFORMATION_H__
+
+#include "iree/hal/drivers/hip/hip_headers.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+typedef struct iree_hal_stream_tracing_context_t
+    iree_hal_stream_tracing_context_t;
+typedef struct iree_hal_hip_event_pool_t iree_hal_hip_event_pool_t;
+typedef struct iree_hal_deferred_work_queue_t iree_hal_deferred_work_queue_t;
+
+typedef struct iree_hal_hip_per_device_information_t {
```

Comment:
We generally try to avoid non-obvious abbreviations (for improving search by making things unambiguous and for helping non-native English speakers) but `information` -> `info` is common enough that it's safe to do. Best to err on the verbose side but here you could save quite a few characters with no loss of readability.

```suggestion
typedef struct iree_hal_hip_per_device_info_t {
```